### PR TITLE
feat(config): Allow reloading on URL config change

### PR DESCRIPTION
**Pull request by powersj**
## Summary

This introduces a new config-url-watch-interval option, which when set will, at each interval, check the Last-Modified header of the file to determine if telegraf should reload.

If the header is not available then the watcher is disabled for the file.

## Checklist

- [x] No AI generated code was used in this PR

## Related issues

fixes: #8730

### CODE REVIEW

File: cmd/telegraf/telegraf.go [hunk 1]
```diff
@@ -149,11 +152,20 @@ func (t *Telegraf) reloadLoop() error {
 			for _, fConfig := range t.configFiles {
 				if _, err := os.Stat(fConfig); err == nil {
 					go t.watchLocalConfig(signals, fConfig)
-				} else {
-					log.Printf("W! Cannot watch config %s: %s", fConfig, err)
```
Comment (srebhan) [inline cmd/telegraf/telegraf.go:153]: Can we keep this warning for local files? I'm not sure if there is a sensible case where you can read the file but not stat it, but if there is, I would like to inform the user that we are actually _not_ watching a file he expects us to watch...

**Review by srebhan — COMMENTED** [review]
Looks good to me, but I've once concern about removing the warning if we cannot watch a file...

File: cmd/telegraf/telegraf.go (hunk 1 above)
Comment (powersj) [reply cmd/telegraf/telegraf.go]: Yep! I've added a check, let me know if that is what you had in mind.

**Review by srebhan — APPROVED** [review]
Awesome. Thanks @powersj!

**Review by DStrand1 — APPROVED** [review]

**Comment by Joris97** [comment]
How does this new function work? Putting "configURLWatchInterval: 1m " under agent or is this already explained somewhere?

**Comment by powersj** [comment]
> How does this new function work? Putting "configURLWatchInterval: 1m " under agent or is this already explained somewhere?

It's a CLI option `--config-url-watch-interval`. `telegraf --help` should list it.

**Pull request merged**